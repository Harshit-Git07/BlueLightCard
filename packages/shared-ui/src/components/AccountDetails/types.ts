export type AccountDetailsProps = {
  accountNumber: string;
  firstName: string;
  lastName: string;
};
