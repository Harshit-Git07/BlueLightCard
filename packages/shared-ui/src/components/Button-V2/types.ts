import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ButtonProps = PropsWithChildren & {
  className?: string;
  type?: 'button' | 'submit';
  disabled?: boolean;
  slim?: boolean;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  newTab?: boolean;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
  size?: ButtonSize;
  'data-testid'?: string;
};

export type ButtonSize = 'XSmall' | 'Small' | 'Large';
