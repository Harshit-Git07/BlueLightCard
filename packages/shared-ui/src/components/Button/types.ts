import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ColorToken = {
  bg: string;
  hover: string;
  focus: string;
  text: string;
  border: string;
};

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
};
