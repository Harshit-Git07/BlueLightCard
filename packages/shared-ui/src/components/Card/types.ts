import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { RefObject } from 'react';

export type CardState = 'default' | 'hover' | 'selected';

export interface CardProps {
  initialCardState?: CardState;
  cardTitle?: string;
  description?: string;
  truncateDescription?: boolean
  buttonTitle?: string;
  imageAlt?: string;
  imageSrc?: string;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  showImage?: boolean;
  showDescription?: boolean;
  showButton?: boolean;
  onClick?: () => void;
  ariaLabel?: string;
  className?: string;
  'data-testid'?: string;
  canHover?: boolean;
}

export interface CardContentProps {
  title: string;
  description?: string;
  showDescription?: boolean;
  centerContent?: boolean;
  ariaLabel?: string;
  descriptionRef: RefObject<HTMLParagraphElement>;
  truncateDescription?: boolean;
}
