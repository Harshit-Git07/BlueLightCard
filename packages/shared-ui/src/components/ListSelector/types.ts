import { ReactNode } from 'react';

export enum ListSelectorState {
  Default = 'default',
  Selected = 'selected',
  Hover = 'hover',
}

export interface ListSelectorProps {
  ariaLabel?: string;
  title?: string;
  state?: ListSelectorState;
  onClick?: () => void;
  tag?: ReactNode;
  description?: ReactNode;
  showTrailingIcon?: boolean;
  className?: string;
  'data-testid'?: string;
}
