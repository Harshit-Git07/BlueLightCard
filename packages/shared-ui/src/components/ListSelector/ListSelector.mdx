import { Meta, Canvas, Controls } from '@storybook/blocks';
import * as ListSelector from './ListSelector.stories';

<Meta of={ListSelector} />

# ListSelector

- [Status](#status)
- [Overview](#overview)
- [Props](#props)
- [State](#state)
- [Implementation](#implementation)
- [Usage Guidelines](#usage-guidelines)
- [Design](#design)


## Overview

OList selectors are used to add more information to the user’s options without adding clutter. When options are explicitly separated into cards, it’s obvious which elements belong to which group and easier to quickly distinguish between list options. It also allows quick editing functionality when used in a form - like allowing user to select their employment status or relevant ID to upload during Sign up on in the MyAccount area.

## Props
<Controls />

###States
List of states can be found below here.

### Default
<Canvas of={ListSelector.Default} />

### Selected
<Canvas of={ListSelector.Selected} />

### Hover
<Canvas of={ListSelector.Hover} />

## Implementation
Usage example using jsx

```jsx
const ListSelector: FC = () => {
  return <></>;
};
```

## Usage Guidelines

➡️ ListSelector may contain icon, on the right side optionally - usually on used on desktop and omitted in default state on mobile due to touch capability.

➡️ ListSelector should be used in a group - to allow user to select ONE from a list of items - not stand alone. Two selectors in a list cannot both be in the selected state at the same time

➡️ Height of list selector should hug the contents.

➡️ Width of list selector usually fills width or parent container but on desktop screen min width is 500px.

➡️ Title in list selector should be kept short.

## Design
Use the `<FigmaEmbed />` component.

Please see the [Figma design](<https://www.figma.com/design/n2NzB4G3OTR74o3dQ4BOwm/My-Account-%26-Sign-up-Component-library?node-id=587-4815&t=yzSssj6IsbylKbTM-4>) for the design of the ListSelector component.
