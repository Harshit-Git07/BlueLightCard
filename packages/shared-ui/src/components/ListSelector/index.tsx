import React, { FC, KeyboardEvent, useEffect, useMemo, useState } from 'react';
import { ListSelectorProps, ListSelectorState } from './types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowRight } from '@fortawesome/pro-regular-svg-icons';
import { faPen } from '@fortawesome/pro-solid-svg-icons';
import { colours, fonts } from '../../tailwind/theme';

const ListSelector: FC<ListSelectorProps> = ({
  className = '',
  ariaLabel,
  title = '',
  state = ListSelectorState.Default,
  onClick,
  tag,
  description,
  showTrailingIcon = true,
}) => {
  const [currentState, setCurrentState] = useState(state);

  useEffect(() => {
    setCurrentState(state);
  }, [state]);

  const handleKeyDown = (e: KeyboardEvent<HTMLButtonElement>) => {
    if (e.key === 'Enter' || e.key === ' ') {
      e.preventDefault();
      handleClick(); // Call handleClick for keyboard events as well
    }
  };

  const handleClick = () => {
    // Set the state to Selected when clicked
    setCurrentState(ListSelectorState.Selected);

    if (onClick) {
      onClick();
    }
  };

  const handleMouseEnter = () => {
    if (currentState !== ListSelectorState.Selected) {
      setCurrentState(ListSelectorState.Hover);
    }
  };

  const handleMouseLeave = () => {
    setCurrentState(state);
  };

  const containerStyles = useMemo(() => {
    switch (currentState) {
      case ListSelectorState.Selected:
        return 'border border-colour-primary dark:border-colour-primary-dark bg-colour-surface-container-light dark:bg-colour-surface-container-dark';
      case ListSelectorState.Hover:
        return 'border border-colour-primary-hover dark:border-colour-primary-dark';
      default:
        return 'border border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark';
    }
  }, [currentState]);

  const iconColour =
    currentState === ListSelectorState.Selected
      ? 'text-colour-primary dark:text-colour-primary-dark'
      : 'text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark';

  return (
    <button
      className={`${className} ${containerStyles} w-full min-w-full sm:min-w-[500px] rounded-[4px] px-[16px] py-[12px] flex items-center justify-between transition-colors border`}
      aria-label={ariaLabel ?? `Button for ${title}`}
      role={onClick ? 'button' : undefined}
      onClick={handleClick}
      onKeyDown={handleKeyDown}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <div className="flex flex-col gap-[4px] items-start w-full text-left">
        <div className="flex flex-row items-center flex-wrap gap-[8px] w-full">
          <h4
            className={`${colours.textOnSurface} ${fonts.bodySemiBold} break-words whitespace-normal`}
          >
            {title}
          </h4>

          {tag}
        </div>

        {description && (
          <p
            className={`${colours.textOnSurfaceSubtle} ${fonts.body} break-words whitespace-normal text-left`}
          >
            {description}
          </p>
        )}
      </div>

      {showTrailingIcon && (
        <FontAwesomeIcon
          className={`${iconColour} h-[20px]`}
          icon={currentState === ListSelectorState.Selected ? faPen : faArrowRight}
        />
      )}
    </button>
  );
};

export default ListSelector;
