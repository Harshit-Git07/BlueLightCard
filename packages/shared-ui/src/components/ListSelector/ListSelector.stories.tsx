import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { ListSelectorState } from './types';
import { faCircleBolt } from '@fortawesome/pro-solid-svg-icons';
import ListSelector from './index';
import Tag from '../Tag';
import { TagState } from '../Tag/types';

const meta: Meta<typeof ListSelector> = {
  title: 'Component System/ListSelector',
  component: ListSelector,
  argTypes: {
    onClick: { action: 'clicked' },
    state: {
      control: 'select',
      options: Object.values(ListSelectorState),
    },
  },
};

export default meta;
type Story = StoryObj<typeof ListSelector>;

const Template: Story = {
  render: (args) => <ListSelector {...args} />,
};
const SUCCESS_TAG = (
  <Tag
    infoMessage="Tag"
    iconLeft={faCircleBolt}
    iconRight={faCircleBolt}
    state={TagState.Success}
  ></Tag>
);

export const Default: Story = {
  ...Template,
  args: {
    title: 'Info Card Title',
    description: 'Info card description text',
    state: ListSelectorState.Default,
    tag: SUCCESS_TAG,
    showTrailingIcon: true,
    onClick: undefined,
  },
};

export const HiddenTag: Story = {
  ...Template,
  args: {
    title: 'Hidden Tag',
    description: 'This is a default list selector',
    state: ListSelectorState.Default,
    tag: undefined,
  },
};

export const HiddenDescription: Story = {
  ...Template,
  args: {
    title: 'Hidden Description',
    state: ListSelectorState.Default,
    tag: undefined,
    description: undefined,
  },
};

export const HiddenTrailingIcon: Story = {
  ...Template,
  args: {
    title: 'Hidden Trailing Icon',
    showTrailingIcon: false,
    tag: undefined,
    description: undefined,
  },
};

export const Selected: Story = {
  ...Template,
  args: {
    ...Default.args,
    title: 'Selected List Selector',
    state: ListSelectorState.Selected,
  },
};

export const Hover: Story = {
  ...Template,
  args: {
    ...Default.args,
    title: 'Hover List Selector',
    state: ListSelectorState.Hover,
  },
};
