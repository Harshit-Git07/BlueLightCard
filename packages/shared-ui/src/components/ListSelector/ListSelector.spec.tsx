import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import ListSelector from './';
import { ListSelectorProps, ListSelectorState } from './types';
import Tag from '../Tag';

describe('ListSelector component', () => {
  let props: ListSelectorProps;

  beforeEach(() => {
    props = {};
  });

  // smoke test
  it('should render component without error', () => {
    const { baseElement } = render(<ListSelector {...props} />);
    expect(baseElement).toBeTruthy();
  });

  it('should render default list selector with title, description, tag and trailing icon  ', () => {
    const component = renderer.create(
      <ListSelector
        title={'Default List Selector'}
        description={'This is a default list selector'}
        tag={<Tag state={'Success'} />}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default list selector with hidden tag', () => {
    const component = renderer.create(
      <ListSelector
        title={'Default List Selector'}
        description={'This is a default list selector'}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default list selector with hidden description', () => {
    const component = renderer.create(
      <ListSelector title={'Default List Selector'} tag={<Tag state={'Success'} />} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default list selector with hidden trailing icon', () => {
    const component = renderer.create(
      <ListSelector
        title={'Default List Selector'}
        description={'This is a default list selector'}
        tag={<Tag state={'Success'} />}
        showTrailingIcon={false}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default list selector with in selected state', () => {
    const component = renderer.create(
      <ListSelector
        state={ListSelectorState.Selected}
        title={'Default List Selector'}
        tag={<Tag state={'Success'} />}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default list selector with in hover state', () => {
    const component = renderer.create(
      <ListSelector
        state={ListSelectorState.Hover}
        title={'Default List Selector'}
        tag={<Tag state={'Success'} />}
        ariaLabel={'List Selector'}
        onClick={() => {}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
