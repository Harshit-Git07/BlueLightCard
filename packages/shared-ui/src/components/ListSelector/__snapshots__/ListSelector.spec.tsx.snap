// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSelector component should render default list selector with hidden description 1`] = `
<button
  aria-label="Button for Default List Selector"
  className="w-full rounded-lg px-4 py-3 flex items-center justify-between transition-colors border border border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark min-w-full sm:min-w-[500px]"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="flex flex-col items-start w-full text-left"
  >
    <div
      className="flex items-center gap-2 mb-1 w-full"
    >
      <h4
        className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold break-words whitespace-normal text-left"
      >
        Default List Selector
      </h4>
      <span>
        <div
          className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-success-bright-light dark:bg-colour-success-bright-dark text-colour-onSuccess-bright-light dark:text-colour-onSuccess-bright-dark border-[0.5px] border-colour-onSuccess-bright-light dark:border-colour-onSuccess-bright-dark rounded"
        >
          <span
            className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
          />
        </div>
      </span>
    </div>
  </div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-right h-5 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    data-icon="arrow-right"
    data-prefix="far"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9 .8s-8.8 24.8 .8 33.9L364.1 232 24 232c-13.3 0-24 10.7-24 24s10.7 24 24 24l340.1 0L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9 .8l176-168z"
      fill="currentColor"
      style={{}}
    />
  </svg>
</button>
`;

exports[`ListSelector component should render default list selector with hidden tag 1`] = `
<button
  aria-label="Button for Default List Selector"
  className="w-full rounded-lg px-4 py-3 flex items-center justify-between transition-colors border border border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark min-w-full sm:min-w-[500px]"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="flex flex-col items-start w-full text-left"
  >
    <div
      className="flex items-center gap-2 mb-1 w-full"
    >
      <h4
        className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold break-words whitespace-normal text-left"
      >
        Default List Selector
      </h4>
    </div>
    <p
      className="text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body break-words whitespace-normal text-left"
    >
      This is a default list selector
    </p>
  </div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-right h-5 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    data-icon="arrow-right"
    data-prefix="far"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9 .8s-8.8 24.8 .8 33.9L364.1 232 24 232c-13.3 0-24 10.7-24 24s10.7 24 24 24l340.1 0L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9 .8l176-168z"
      fill="currentColor"
      style={{}}
    />
  </svg>
</button>
`;

exports[`ListSelector component should render default list selector with hidden trailing icon 1`] = `
<button
  aria-label="Button for Default List Selector"
  className="w-full rounded-lg px-4 py-3 flex items-center justify-between transition-colors border border border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark min-w-full sm:min-w-[500px]"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="flex flex-col items-start w-full text-left"
  >
    <div
      className="flex items-center gap-2 mb-1 w-full"
    >
      <h4
        className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold break-words whitespace-normal text-left"
      >
        Default List Selector
      </h4>
      <span>
        <div
          className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-success-bright-light dark:bg-colour-success-bright-dark text-colour-onSuccess-bright-light dark:text-colour-onSuccess-bright-dark border-[0.5px] border-colour-onSuccess-bright-light dark:border-colour-onSuccess-bright-dark rounded"
        >
          <span
            className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
          />
        </div>
      </span>
    </div>
    <p
      className="text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body break-words whitespace-normal text-left"
    >
      This is a default list selector
    </p>
  </div>
</button>
`;

exports[`ListSelector component should render default list selector with in hover state 1`] = `
<button
  aria-label="List Selector"
  className="w-full rounded-lg px-4 py-3 flex items-center justify-between transition-colors border border border-colour-primary-hover dark:border-colour-primary-dark min-w-full sm:min-w-[500px]"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
>
  <div
    className="flex flex-col items-start w-full text-left"
  >
    <div
      className="flex items-center gap-2 mb-1 w-full"
    >
      <h4
        className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold break-words whitespace-normal text-left"
      >
        Default List Selector
      </h4>
      <span>
        <div
          className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-success-bright-light dark:bg-colour-success-bright-dark text-colour-onSuccess-bright-light dark:text-colour-onSuccess-bright-dark border-[0.5px] border-colour-onSuccess-bright-light dark:border-colour-onSuccess-bright-dark rounded"
        >
          <span
            className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
          />
        </div>
      </span>
    </div>
  </div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-right h-5 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    data-icon="arrow-right"
    data-prefix="far"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9 .8s-8.8 24.8 .8 33.9L364.1 232 24 232c-13.3 0-24 10.7-24 24s10.7 24 24 24l340.1 0L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9 .8l176-168z"
      fill="currentColor"
      style={{}}
    />
  </svg>
</button>
`;

exports[`ListSelector component should render default list selector with in selected state 1`] = `
<button
  aria-label="Button for Default List Selector"
  className="w-full rounded-lg px-4 py-3 flex items-center justify-between transition-colors border border border-colour-primary dark:border-colour-primary-dark bg-colour-surface-container-light dark:bg-colour-surface-container-dark min-w-full sm:min-w-[500px]"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="flex flex-col items-start w-full text-left"
  >
    <div
      className="flex items-center gap-2 mb-1 w-full"
    >
      <h4
        className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold break-words whitespace-normal text-left"
      >
        Default List Selector
      </h4>
      <span>
        <div
          className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-success-bright-light dark:bg-colour-success-bright-dark text-colour-onSuccess-bright-light dark:text-colour-onSuccess-bright-dark border-[0.5px] border-colour-onSuccess-bright-light dark:border-colour-onSuccess-bright-dark rounded"
        >
          <span
            className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
          />
        </div>
      </span>
    </div>
  </div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-pen h-5 text-colour-primary dark:text-colour-primary-dark"
    data-icon="pen"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M362.7 19.3L314.3 67.7 444.3 197.7l48.4-48.4c25-25 25-65.5 0-90.5L453.3 19.3c-25-25-65.5-25-90.5 0zm-71 71L58.6 323.5c-10.4 10.4-18 23.3-22.2 37.4L1 481.2C-1.5 489.7 .8 498.8 7 505s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L421.7 220.3 291.7 90.3z"
      fill="currentColor"
      style={{}}
    />
  </svg>
</button>
`;

exports[`ListSelector component should render default list selector with title, description, tag and trailing icon   1`] = `
<button
  aria-label="Button for Default List Selector"
  className="w-full rounded-lg px-4 py-3 flex items-center justify-between transition-colors border border border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark min-w-full sm:min-w-[500px]"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="flex flex-col items-start w-full text-left"
  >
    <div
      className="flex items-center gap-2 mb-1 w-full"
    >
      <h4
        className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold break-words whitespace-normal text-left"
      >
        Default List Selector
      </h4>
      <span>
        <div
          className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-success-bright-light dark:bg-colour-success-bright-dark text-colour-onSuccess-bright-light dark:text-colour-onSuccess-bright-dark border-[0.5px] border-colour-onSuccess-bright-light dark:border-colour-onSuccess-bright-dark rounded"
        >
          <span
            className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
          />
        </div>
      </span>
    </div>
    <p
      className="text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body break-words whitespace-normal text-left"
    >
      This is a default list selector
    </p>
  </div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-right h-5 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    data-icon="arrow-right"
    data-prefix="far"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9 .8s-8.8 24.8 .8 33.9L364.1 232 24 232c-13.3 0-24 10.7-24 24s10.7 24 24 24l340.1 0L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9 .8l176-168z"
      fill="currentColor"
      style={{}}
    />
  </svg>
</button>
`;
